### PR TITLE
Add Run_Gemma3.md and training examples for Gemma3

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Add an instruction readme for Gemma3, and training examples for Gemma3-4b/12b/27b models. Pre-converted checkpoints path are included for Googlers.

# Tests

Trainings/fine-tunings have been tested through workloads for all three models ([4b](https://pantheon.corp.google.com/logs/query;cursorTimestamp=2025-03-24T06:05:54.160589234Z;duration=P1D;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-prod-env-multipod%22%0Aresource.labels.location%3D%22us-west1%22%0Aresource.labels.cluster_name%3D%22bodaborg-v6e-256-ts%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22htg-gemma3-4b-finetune-slice-job-0-0-%22%20severity%3E%3DDEFAULT%0Atimestamp%3D%222025-03-24T06:05:54.160589234Z%22%0AinsertId%3D%220y38sw71mjdi1j1p%22;storageScope=project?project=tpu-prod-env-multipod), [12b](https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-prod-env-multipod%22%0Aresource.labels.location%3D%22us-west1%22%0Aresource.labels.cluster_name%3D%22bodaborg-v6e-256-ts%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22htg-gemma3-12b-finetune-slice-job-0-0-%22%20severity%3E%3DDEFAULT;storageScope=project;cursorTimestamp=2025-03-24T06:10:55.725463317Z;duration=P1D?e=13802955&mods=allow_workbench_image_override&project=tpu-prod-env-multipod), [27b](https://pantheon.corp.google.com/logs/query;cursorTimestamp=2025-03-24T17:49:03.417790027Z;duration=P1D;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-prod-env-multipod%22%0Aresource.labels.location%3D%22us-west1%22%0Aresource.labels.cluster_name%3D%22bodaborg-v6e-256-ts%22%0Aresource.labels.namespace_name%3D%22default%22%0Alabels.k8s-pod%2Fjobset_sigs_k8s_io%2Fjobset-name%3D%22htg-gemma3-27b-finetune%22%20severity%3E%3DDEFAULT%0Atimestamp%3D%222025-03-24T17:48:45.015063739Z%22%0AinsertId%3D%22m1n6ebs73c8e0tc0%22;storageScope=project?project=tpu-prod-env-multipod)).

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: end_to_end/tpu/gemma3/Run_Gemma3.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!--
+ Copyright 2023 Google LLC
```
Comment (gagika) [inline end_to_end/tpu/gemma3/Run_Gemma3.md:2]: please update to 2025.

Cloud you please add some examples pretraining/fine tuning and decode here similar to this:

https://github.com/AI-Hypercomputer/maxtext/blob/ds_support/end_to_end/tpu/deepseek/Run_DeepSeek.md

**Review by gagika — COMMENTED** [review]
Thanks, looks good, only 1 comment

File: end_to_end/tpu/gemma3/27b/test_gemma3.sh [hunk 2]
```diff
@@ -47,3 +47,29 @@ python MaxText/decode.py MaxText/configs/base.yml tokenizer_path=assets/tokenize
...
+python MaxText/train.py MaxText/configs/base.yml model_name=$MODEL_NAME \
+  base_output_directory=${BASE_OUTPUT_DIRECTORY} dataset_path=${DATASET_PATH} \
+  load_parameters_path=${CONVERTED_CHECKPOINT} \
+  tokenizer_path=assets/tokenizer.gemma3 per_device_batch_size=2 \
+  run_name=$FINETUNE_RUN_NAME max_target_length=8192 steps=15 enable_checkpointing=true \
+  sharding_tolerance=0.03 profiler=xplane \
+  remat_policy=full ici_fsdp_transpose_parallelism=256 ici_fsdp_parallelism=-1 \
+  reuse_example_batch=1 dataset_type=synthetic gcs_metrics=true \
+  attention='flash' sa_block_q=2048 sa_block_q_dkv=2048 sa_block_q_dq=2048
+
```
Comment (gagika) [inline end_to_end/tpu/gemma3/27b/test_gemma3.sh:65]: is this for performance run for Trillium, if so, could you please move it to 

MaxText/configs/trillium/gemma3_27b.sh

here you can just keep generic pretraining/finetuning similar to https://github.com/AI-Hypercomputer/maxtext/tree/ds_support/end_to_end/tpu/llama3.1/70b

File: end_to_end/tpu/gemma3/Run_Gemma3.md (hunk 1 above)
Comment (hengtaoguo) [reply end_to_end/tpu/gemma3/Run_Gemma3.md]: Thanks! Added some example instructions about pretrain/finetune/decode in the Run_Gemma3.md page.

File: end_to_end/tpu/gemma3/27b/test_gemma3.sh (hunk 2 above)
Comment (hengtaoguo) [reply end_to_end/tpu/gemma3/27b/test_gemma3.sh]: Yes this was for Trillium perf run. I moved this piece to "MaxText/configs/trillium/gemma3_27b.sh", and simplified the training examples in test_gemma3.sh for 4b/12b/27b models. Thanks for the suggestions!

**Review by gagika — APPROVED** [review]
Thanks!

**Review by gobbleturk — APPROVED** [review]
